orthoinference/src/main/java/org/reactome/orthoinference/SkipInstanceChecker.java

```diff
@@ -101,9 +107,111 @@ public static boolean checkIfInstanceShouldBeSkipped(GKInstance reactionInst) th
+		// Reactions with no proteins/EWAS (Total = 0) are not inferred.
+		List<Integer> reactionProteinCounts = ProteinCountUtility.getDistinctProteinCounts(reactionInst);
+		int reactionTotalProteinCounts = reactionProteinCounts.get(0);
+		if (reactionTotalProteinCounts > 0) {
+			logger.info("Total protein count for RlE: " + reactionTotalProteinCounts);
+			eligibleCount++;
+			String eligibleEventName = reactionInst.getAttributeValue(DB_ID).toString() + "\t" + reactionInst.getDisplayName() + "\n";
+			Files.write(Paths.get(eligibleFilehandle), eligibleEventName.getBytes(), StandardOpenOption.APPEND);
+			// Checks that ReactionlikeEvents will be fully inferrable before attempting inference
+			if (!reactionComponentsAreInferrable(reactionInst)) {
+				return true;
+			}
+		} else {
+				logger.info("No distinct proteins found in instance -- terminating inference for " + reactionInst);
+				return true;
+			}
 		return false;
 	}
-	
+
+	/**
+	 * 	Each input, output, and catalyst is screened to verify that the Reaction will be inferred.
+	 * 	This prevents the majority of orphan PEs that were being created during orthoinference.
+	 * 	Only some EntitySets are still orphaned due to the complexity behind attempting to screen them
+	 * @param reactionInst -- GKInstance that will be screened
+	 * @return -- Boolean is returned that indicates if Reaction is fully inferrable or not
+	 * @throws Exception
+	 */
+	private static boolean reactionComponentsAreInferrable(GKInstance reactionInst) throws Exception {
+		// First gather all inputs, outputs and the PEs in catalyst activities
+		// Inputs/Outputs/CatalystPEs need to be stored in seperate collections. At time of writing, having it all stored in
+		// the same collection causes outputs to be inferred in both inputs and outputs during the actual inference -- not ideal
+		Collection<GKInstance> reactionInputs = reactionInst.getAttributeValuesList(input);
+		Collection<GKInstance> reactionOutputs = reactionInst.getAttributeValuesList(output);
+		Collection<GKInstance> reactionCatalystPEs = new ArrayList<>();
+		Collection<GKInstance> reactionCatalysts = reactionInst.getAttributeValuesList(catalystActivity);
+		for (GKInstance reactionCatalyst : reactionCatalysts) {
+			GKInstance catalystPE = (GKInstance) reactionCatalyst.getAttributeValue(physicalEntity);
+			if (catalystPE != null) {
+				reactionCatalystPEs.add(catalystPE);
+			}
+		}
+		// Screen inputs
+		for (GKInstance reactionInput : reactionInputs) {
+			if (!componentIsInferrable(reactionInput)) {
+				return false;
+			}
+		}
+		// Screen outputs
+		for (GKInstance reactionOutput : reactionOutputs) {
+			if (!componentIsInferrable(reactionOutput)) {
+				return false;
+			}
+		}
+		// Screen catalyst PhysicalEntities
+		for (GKInstance reactionCatalystPE : reactionCatalystPEs) {
+			if (!componentIsInferrable(reactionCatalystPE)) {
+				return false;
+			}
+		}
+		return true;
```

Comment:
The only thing is the `componentIsInferrable` exception would need to be handled in that method rather than thrown.  If it's practical to handle the exception in the method it gets thrown rather than throwing it to the method caller it can help reduce the methods which throw exceptions.

Reply:
Yes, `allMatch` is a short-circuiting operation: https://docs.oracle.com/javase/8/docs/api/java/util/stream/Stream.html#allMatch-java.util.function.Predicate-

Reply:
One way to simplify this might be to add all of the inputs, outputs, PEs into a big list and then just use one loop:
```
List<GKInstance> bigList = new ArrayList<>();
bigList.addAll(reactionInputs);
bigList.addAll(reactionCatalystPEs);
bigList.addAll(reactionOutputs);

for (GKInstance instance : bigList) {
    if (!componentIsInferrable(instance)) {
        return false;
    }
}
return true;
```

Reply:
This still has `return` in the middle of the function, but only in two places, and the code is smaller so it's not as bad ;)